lib/vector-common/src/shutdown.rs

```diff
@@ -336,7 +347,7 @@ mod test {
         let mut shutdown = SourceShutdownCoordinator::default();
         let id = ComponentKey::from("test");
 
-        let (shutdown_signal, _) = shutdown.register_source(&id);
+        let (shutdown_signal, _) = shutdown.register_source(&id, false);
```

Comment:
Would there be a good way of testing that one set shuts down before the other does?